core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -397,23 +400,37 @@ object ConsumerGroupCommand extends Logging {
           case None =>
             None
           case Some(consumers) =>
-            if (consumers.isEmpty)
-              Some(Array[PartitionAssignmentState]())
-            else {
-              val consumer = getConsumer()
-              Some(consumers.sortWith(_.assignment.size > _.assignment.size).flatMap { consumerSummary =>
-                val topicPartitions = consumerSummary.assignment.map(tp => TopicAndPartition(tp.topic, tp.partition))
-                val partitionOffsets = topicPartitions.flatMap { topicPartition =>
-                  Option(consumer.committed(new TopicPartition(topicPartition.topic, topicPartition.partition))).map { offsetAndMetadata =>
-                    topicPartition -> offsetAndMetadata.offset
-                  }
-                }.toMap
-                collectConsumerAssignment(group, Some(consumerGroupSummary.coordinator), topicPartitions,
-                  partitionOffsets.get, Some(s"${consumerSummary.consumerId}"), Some(s"${consumerSummary.host}"),
-                  Some(s"${consumerSummary.clientId}"))
-              })
-            }
-        }
+            var assignedTopicPartitions = Array[TopicPartition]()
+            val offsets = adminClient.listGroupOffsets(group)
+            val rowsWithConsumer =
+              if (offsets.isEmpty)
+                List[PartitionAssignmentState]()
+              else
+                consumers.sortWith(_.assignment.size > _.assignment.size).flatMap { consumerSummary =>
+                  val topicPartitions = consumerSummary.assignment.map(tp => TopicAndPartition(tp.topic, tp.partition))
+                  assignedTopicPartitions = assignedTopicPartitions ++ consumerSummary.assignment
+                  val partitionOffsets: Map[TopicAndPartition, Option[Long]] = consumerSummary.assignment.map { topicPartition =>
+                    TopicAndPartition(topicPartition.topic, topicPartition.partition) -> {
+                      if (offsets.keySet.contains(topicPartition))
```

Comment:
Right, and with the change to return type of `listGroupOffsets` this will become `offsets.get(topicPatition)`. Thanks.